doc/evp_skey.md

```diff
@@ -0,0 +1,99 @@
+EVP_SKEY Design
+===============
+
+We currently presume that symmetric keys are represnted as octet buffer. It may
+be not relevant for the situation of hardware-backed providers, such as
+PKCS#11 or TPM.
+
+This is a design of using opaque usage in symmetric encryption.
+
+Key management
+--------------
+
+The provider operating EVP_SKEY objects has to implement a separate key
+management to deal with opaque object. We already have MAC key management and
```

Comment:
MAC key management?

Reply:
https://github.com/openssl/openssl/blob/master/providers/implementations/keymgmt/mac_legacy_kmgmt.c

Reply:
Oh...  legacy stuff that we made so those who used the MAC-as-an-EVP_PKEY horror wouldn't have a too rough awakening.  Yeahok